src/Uno.UI/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -0,0 +1,358 @@
+		}
+
+		private void OnContainerContentChanging(object sender, ContainerContentChangingEventArgs args)
+		{
+			//	m_containerContentChangedSource(sender.as< winrt::ListView > (), args);
+		}
+
+		private void OnNodeExpanding(TreeViewNode sender, EventArgs args)
+		{
+			var treeViewExpandingEventArgs = new TreeViewExpandingEventArgs(sender);
+
+			if (ListControl != null)
+			{
+				if (ContainerFromNode(sender) is TreeViewItem expandingTvi)
+				{
+					if (!expandingTvi.IsExpanded)
+					{
+						expandingTvi.IsExpanded = true;
+					}
+
+					//Update TemplateSettings properties
+					var templateSettings = expandingTvi.TreeViewItemTemplateSettings;
+					templateSettings.ExpandedGlyphVisibility = Visibility.Visible;
+					templateSettings.CollapsedGlyphVisibility = Visibility.Collapsed;
+				}
+				//TODO: m_expandingEventSource(*this, *treeViewExpandingEventArgs);
+			}
+		}
+
+		private void OnNodeCollapsed(TreeViewNode sender, EventArgs args)
+		{
+			var treeViewCollapsedEventArgs = new TreeViewCollapsedEventArgs(sender);
+
+			if (ListControl != null)
+			{
+				if (ContainerFromNode(sender) is TreeViewItem collapsedTvi)
+				{
+					//Update TVI properties
+					if (collapsedTvi.IsExpanded)
+					{
+						collapsedTvi.IsExpanded = false;
+					}
+
+					//Update TemplateSettings properties
+					var templateSettings = collapsedTvi.TreeViewItemTemplateSettings;
+					templateSettings.ExpandedGlyphVisibility = Visibility.Collapsed;
+					templateSettings.CollapsedGlyphVisibility = Visibility.Visible;
+				}
+				//TODO: m_collapsedEventSource(*this, *treeViewCollapsedEventArgs);
+			}
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
+
+			if (property == SelectionModeProperty && ListControl != null)
+			{
+				TreeViewSelectionMode value = SelectionMode;
+				switch (value)
+				{
+					case TreeViewSelectionMode.None:
+						{
+							ListControl.SelectionMode = ListViewSelectionMode.None;
+							UpdateItemsSelectionMode(false);
+						}
+						break;
+
+					case TreeViewSelectionMode.Single:
+						{
+							ListControl.SelectionMode = ListViewSelectionMode.Single;
+							UpdateItemsSelectionMode(false);
+						}
+						break;
+
+					case TreeViewSelectionMode.Multiple:
+						{
+							ListControl.SelectionMode = ListViewSelectionMode.None;
+							UpdateItemsSelectionMode(true);
+						}
+						break;
+
+				}
+			}
+			else if (property == ItemsSourceProperty)
+			{
+				m_rootNode.IsContentMode = true;
+
+				if (ListControl != null)
+				{
+					var viewModel = ListControl.ListViewModel;
+					viewModel.IsContentMode = true;
+				}
+
+				m_rootNode.ItemsSource = ItemsSource;
+			}
+		}
+
+		private void OnListControlDragItemsStarting(object sender, DragItemsStartingEventArgs args)
+		{
+			var treeViewArgs = new TreeViewDragItemsStartingEventArgs(args);
+			//TODO: m_dragItemsStartingEventSource(*this, *treeViewArgs);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)